### PR TITLE
Fix undefined idbDatabase.objectStoreNames

### PR DESCRIPTION
**Pull request by rwwagner90**
This was throwing errors when `idbDatabase.objectStoreNames` was undefined because it is not iterable in `Array.from(idbDatabase.objectStoreNames)`

### CODE REVIEW

**Comment by rwwagner90** [comment]
@Polarisation can we have tests run on Travis? Not sure if these changes break anything.

**Comment by rwwagner90** [comment]
I updated the travis config, but I think we need some travis settings changes to get test status reported on PRs. We seem to have some test failures here, but I don't really understand the tests. Could you help me figure the failures out please?

**Comment by Polarisation** [comment]
The failure seems to be related to passing a `Set` to `.transaction` which requires an `Array`.

Do you know under what circumstances `.objectStoreNames` can be undefined?

**Comment by rwwagner90** [comment]
@Polarisation `idbDatabase.objectStoreNames` is undefined when you don't have any object stores.

**Comment by rwwagner90** [comment]
@Polarisation I just fixed everything to make sure we're using an array. Let me know how this looks now.

**Comment by rwwagner90** [comment]
@Polarisation I am getting a ton of errors in my app, I think around `transaction.objectStore(storeName).clear().onsuccess = () => {`

I see `IDBRequest.Array.from.forEach.t.objectStore.clear.onsuccess` in Sentry and the error I get is:

`undefined is not iterable (cannot read property Symbol(Symbol.iterator))`

Any ideas what would be causing this?

**Comment by Polarisation** [comment]
@rwwagner90 I've not seen that, sorry.

It would be helpful to add a unit test for the undefined case (I already added one for empty array)

File: index.js [hunk 1]
```diff
@@ -7,18 +7,18 @@
  */
 function exportToJsonString(idbDatabase, cb) {
   const exportObject = {};
-  const size = new Set(idbDatabase.objectStoreNames).size;
+  let objectStoreNames = new Set(idbDatabase.objectStoreNames);
```
Comment (Polarisation) [inline index.js:10]: Please could we stick with `const`? We can use separate variables for the Set and the Array to avoid confusion.

File: index.js [hunk 2]
```diff
@@ -52,47 +52,49 @@ function exportToJsonString(idbDatabase, cb) {
  * @return {void}
  */
 function importFromJsonString(idbDatabase, jsonString, cb) {
-  const objectStoreNames = new Set(idbDatabase.objectStoreNames);
+  let objectStoreNames = new Set(idbDatabase.objectStoreNames);
```
Comment (Polarisation) [inline index.js:55]: const

File: index.js [hunk 3]
```diff
@@ -105,26 +107,29 @@ function importFromJsonString(idbDatabase, jsonString, cb) {
  * @return {void}
  */
 function clearDatabase(idbDatabase, cb) {
-  const size = new Set(idbDatabase.objectStoreNames).size;
+  let objectStoreNames = new Set(idbDatabase.objectStoreNames);
```
Comment (Polarisation) [inline index.js:110]: const

**Review by Polarisation — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (rwwagner90) [reply index.js:10]: I'm not sure I see the benefit? It would be less changes to just keep it as is, instead of requiring separate variables.

**Comment by rwwagner90** [comment]
@Polarisation I'm still not sure I agree with the `const` being any cleaner than `let`, but I have made the changes now. Can we please get this merged and released?

**Comment by Polarisation** [comment]
I added a unit test for `.objectStoreNames` being `undefined`. It passes even with the last release so I'm not really clear what the problem was.

But I'll release these changes since they don't break anything.
